python/taichi/lang/matrix.py

```diff
@@ -590,6 +584,47 @@ def rotation2d(alpha):
         return ti.Matrix([[ti.cos(alpha), -ti.sin(alpha)],
                           [ti.sin(alpha), ti.cos(alpha)]])
 
+    @staticmethod
+    def var(n, m, dt, shape):
+        import taichi as ti
+        return ti.Matrix(n=n, m=m, dt=dt, shape=shape)
+
+    @staticmethod
+    def rows(rows):
+        import taichi as ti
+        mat = ti.Matrix()
+        mat.n = len(rows)
+        if isinstance(rows[0], Matrix):
+            for row in rows:
+                assert row.m == 1, "inputs must be vectors, ie. size n by 1"
+                assert row.n == rows[
+                    0].n, "input vectors must be the same shape"
+            mat.m = rows[0].n
+            # l-value copy:
+            mat.entries = [row(i) for row in rows for i in range(row.n)]
+        elif isinstance(rows[0], list):
+            for row in rows:
+                assert len(row) == len(
+                    rows[0]), "input lists must be the same shape"
+            mat.m = len(rows[0])
+            # l-value copy:
+            mat.entries = [x for row in rows for x in row]
+        else:
+            raise Exception(
+                "cols/rows must be list of lists or lists of vectors")
+        return mat
+
+    @staticmethod
+    def cols(cols):
+        import taichi as ti
+        return ti.Matrix.rows(cols).transpose()
+
+    @staticmethod
+    def empty(n, m, dt=None):
```

Comment:
Mark: btw, I find that `dt` seems only work for tensor-of-matrices, not for matrices. `self.dt` might needs refactor.